stan/math/opencl/kernels/normal_id_glm_lpdf.hpp

```diff
@@ -0,0 +1,162 @@
+#ifndef STAN_MATH_OPENCL_KERNELS_NORMAL_ID_GLM_LPDF_HPP
+#define STAN_MATH_OPENCL_KERNELS_NORMAL_ID_GLM_LPDF_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+
+// \cond
+static const char* normal_id_glm_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * GPU implementation of Generalized Linear Model (GLM)
+     * with Normal distribution and identity link function.
+     *
+     * Must be run with at least N threads and local size equal to LOCAL_SIZE_.
+     * @param[in] y_glob vector parameter
+     * @param[in] x design matrix
+     * @param[in] alpha intercept (in log odds)
+     * @param[in] beta weight vector
+     * @param[in] sigma_glob (Sequence of) scale parameters for the normal
+     * @param[out] mu_derivative_glob intermediate variable used in the model
+     * @param[out] mu_derivative_sum partially summed mu_derivative_glob (1
+     * value per work group)
+     * @param[out] y_minus_mu_over_sigma_squared_sum intermediate variable used
+     * in the model
+     * @param[out] sigma_derivative derivative with respect to sigma
+     * @param[out] log_sigma_sum partially summed logarithm of sigma (1 value
+     * per work group)
+     * @param N number of cases
+     * @param M number of attributes
+     * @param is_alpha_vector 0 or 1 - whether alpha is a vector (alternatively
+     * it is a scalar)
+     * @param is_sigma_vector 0 or 1 - whether sigma is a vector (alternatively
+     * it is a scalar)
+     * @param need_mu_derivative interpreted as boolean - whether mu_derivative
+     * needs to be computed
+     * @param need_mu_derivative_sum interpreted as boolean - whether
+     * mu_derivative_sum needs to be computed
+     * @param need_sigma_derivative interpreted as boolean - whether
+     * sigma_derivative needs to be computed
+     * @param need_log_sigma_sum interpreted as boolean - whether log_sigma_sum
+     * needs to be computed
+     */
+    __kernel void normal_id_glm(
+        const __global double* y_glob, const __global double* x,
+        const __global double* alpha, const __global double* beta,
+        const __global double* sigma_glob, __global double* mu_derivative_glob,
+        __global double* mu_derivative_sum,
+        __global double* y_minus_mu_over_sigma_squared_sum,
+        __global double* sigma_derivative, __global double* log_sigma_sum,
+        const int N, const int M, const int is_alpha_vector,
+        const int is_sigma_vector, const int need_mu_derivative,
+        const int need_mu_derivative_sum, const int need_sigma_derivative,
+        const int need_log_sigma_sum) {
```

Comment:
I'd rather have out parameters first

Reply:
And for names 


y_minus_mu_over_sigma_squared_sum: 
- y_scaled
  - Doc: The (independent variable) scaled by it's mean and variance.

sigma_derivative
- sigma_d
  - Doc: Derivative with respect to sigma

Reply:
> I'd rather have out parameters first

Hmm, so far we have been inconsistent. I will change it here, but there are still other kernels that have inputs before outputs.

Reply:
> And for names

I think having names consistent with CPU implementation is preferable. Anyway I think whole `derivative` is not too loong for variable names. `y_minus_mu_over_sigma_squared_sum` could be shortened, but should be kept consistent with CPU implementation. 

I tried to also keep docs consistent. Imo they could also be improved. However your suggestion for `y_minus_mu_over_sigma_squared_sum` is wrong. I tried to incorporate the idea of your suggestion anyway.

Reply:
> but there are still other kernels that have inputs before outputs.

Let's make a separate PR to correct those later. It's nice to be consistent with in/out order

> I think having names consistent with CPU implementation is preferable. Anyway I think whole derivative is not too loong for variable names. y_minus_mu_over_sigma_squared_sum could be shortened, but should be kept consistent with CPU implementation.

Yeah maybe derivative is fine. I think it's fine to change the names in both here and the CPU implimentation

> I tried to also keep docs consistent. Imo they could also be improved. However your suggestion for y_minus_mu_over_sigma_squared_sum is wrong. I tried to incorporate the idea of your suggestion anyway.

In R calling `scale` is doing `(y - mean(y)) / sd(y)`. Is it not common parlance to call that the scaled value?